radicale/storage/multifilesystem/__init__.py

```diff
@@ -83,6 +84,15 @@ def _atomic_write(self, path, mode="w", newline=None, sync_directory=True,
                 raise RuntimeError("Fsync'ing file %r failed: %s" %
                                    (path, e)) from e
             tmp.close()
+            if os.path.exists(path):
+                os.chmod(tmp.name, os.stat(path).st_mode)
```

Comment:
Multiple threads/processes can change the filesystem in parallel. The file might get deleted after the ``exists`` check.